### PR TITLE
Feature tables

### PR DESCRIPTION
**Pull request by Spground**

### CODE REVIEW

File: src/app/admin/modules/data-management/components/table-export/table-export.component.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    doTableExport() {
+        const url = `${environment.API_URL}/data-management/table-export`;
+        this.tableExportService.exportTable(url).subscribe(
+            data => {
+                const path = `${environment.HOST}${data['url']}`;
+                this.windowService.nativeWindow.open(path);
+                this.snackBar.open('导出成功', '确定', {duration: 3000});
+            },
+            (error: HttpErrorResponse) => {
+                console.log(error);
```
Comment (Time1ess) [inline src/app/admin/modules/data-management/components/table-export/table-export.component.ts]: Remove `console.log`

File: src/app/shared/services/logger.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+describe('LoggerService Test', () => {
+    const testLoggerService: LoggerService = new LoggerService();
+    beforeEach(() => {
+        spyOn(console, 'log');
+        spyOn(console, 'error');
+    });
+
+    it('should be called', () => {
+        testLoggerService.log('TESTing LOG');
+        if (!environment.production) {
```
Comment (Time1ess) [inline src/app/shared/services/logger.service.spec.ts]: In these unit tests, `environment.production` would never be `true`, so the if condition is unnecessary. But this way you can't test whether there will be logs in when `environment.production` is `true`, consider adding mocks to test this service in a production mode.

File: src/environments/environment.interface.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ export interface Environment {
   production: boolean;
   /** The API server address. */
   API_URL: string;
+  HOST: string;
```
Comment (Time1ess) [inline src/environments/environment.interface.ts]: What's this property for? Please add comments.

**Review by Time1ess — CHANGES_REQUESTED** [review]

File: src/app/shared/services/logger.service.spec.ts (hunk 2 above)
Comment (Spground) [reply src/app/shared/services/logger.service.spec.ts]: [https://stackoverflow.com/questions/49091393/how-to-mock-environment-files-import-in-unit-tests](url)
貌似`enviroment.ts`不能被mock，所以这里考虑使用一个`enviroment service`来抽象`enviroment.ts`，同时也能方便后续单测的mock，so any suggestions about this? @Time1ess

Comment (Time1ess) [reply src/app/shared/services/logger.service.spec.ts]: An `EnvironmentService` LGTM.

File: src/environments/environment.interface.ts (hunk 3 above)
Comment (Time1ess) [reply src/environments/environment.interface.ts]: I mean it seems that there is no need to keep this field anymore.

File: src/app/shared/services/environment.service.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import { environment } from 'src/environments/environment';
+import { Environment } from 'src/environments/environment.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EnvironmentService {
```
Comment (Time1ess) [inline src/app/shared/services/environment.service.ts:8]: If `EnvironmentService` has no need to add tests, you might need to exclude it from coverage report by adding it in the `codeCoverageExclude` section in `angular.json`.

**Review by Time1ess — APPROVED** [review]

**Pull request merged**